SSPROD-26997: Changes to support key alias and global deployment

mdkulkarni15 opened the pull request:
<!--
Thank you for your contribution!

## Testing your PR

You can pinpoint the pr changes as terraform module source with following format

```
source                    = "github.com/draios/terraform-aws-secure-for-cloud//examples/organizational-ecs?ref=<BRANCH_NAME>"
```


## General recommendations
Check contribution guidelines at https://github.com/draios/terraform-aws-secure-for-cloud/blob/master/CONTRIBUTE.md#contribution-checklist

For a cleaner PR make sure you follow these recommendations:
- Review modified files and delete small changes that were not intended and maybe slip the commit.
- Use Pull Request Drafts for visibility on Work-In-Progress branches and use them on daily mob/pairing for team review
- Unless an external revision is desired, in order to validate or gather some feedback, you are free to merge as long as **validation checks are green-lighted**

## Checklist

- [ ] If `test/fixtures/*/main.tf` files are modified, update:
    - [ ] the snippets in the README.md file under root folder.
    - [ ] the snippets in the README.md file for the corresponding example.
- [ ] If `examples` folder are modified, update:
    - [ ] README.md file with pertinent changes.
    - [ ] `test/fixtures/*/main.tf` in case the snippet needs modifications.
- [ ] If any architectural change has been made, update the diagrams.

-->

mdkulkarni15 committed:
SSPROD-26789: Changes to support key alias and global resource deployment

ravinadhruve10 commented on modules/services/agentless-scanning/variables.tf:
```diff
@@ -26,3 +26,13 @@ variable "tags" {
   }
 }
+variable "deploy_global_resources" {
+  description = "(Optional) Set this field to 'true' to deploy EventBridge to an AWS Organization (Or specific OUs)"
```
Is this description as expected for agentless scanning feature?

haresh-suresh replied:
Yeah this is definitely not right. We should just substitute, `Eventbridge` with `Agentless Scanning`.
cc: @nkraemer-sysdig

ravinadhruve10 replied:
Done, fixed the description.

ravinadhruve10 reviewed (commented):
As per my understanding, rest of the change lgtm.

noahkraemer reviewed (changes requested):
There are a few missing pieces here:

1) The `deploy_global_resources` flag needs to be set on other global resources (iam policies etc)
2) We need to create a [kms_replica_key](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/kms_replica_key) in each region
3) Variable description needs fixing as Ravina pointed out

ravinadhruve10 committed:
SSPROD-26997: Create kms_replica_key and add checks for deploy_global_resources

Change summary:
---------------
1. Added creation of kms replica key.
2. Added creating kms alias for primary and replica key.
3. Added checks for deploy_global_resources flag in all data and resource instances.
4. Fixed the variable description for deploy_global_resources variable.

Testing done:
-------------
Validated the terrform resources generation using terraform init, validate and plan.

ravinadhruve10 commented:
Added the relevant changes for resolving the missing pieces mentioned above.

ravinadhruve10 committed:
SSPROD-26997: Fix terraform fmt

nkraemer-sysdig commented on modules/services/agentless-scanning/main.tf:
```diff
@@ -231,15 +243,25 @@ data "aws_iam_policy_document" "key_policy" {
+resource "aws_kms_replica_key" "agentless_replica" {
+  count = var.deploy_global_resources ? 0 : 1
+
+  description             = "Multi-Region replica key"
```
Maybe add a `Sysdig` to this description?

ravinadhruve10 replied:
yes, makes sense. Updated and fixed in next commit change.

noahkraemer reviewed (dismissed):
Looking good!

ravinadhruve10 committed:
SSPROD-26997: Update kms_replica_key description
